Sources/NIOPosix/Bootstrap.swift

```diff
@@ -3169,6 +3169,324 @@ public final class NIOPipeBootstrap {
     }
 }
 
+// MARK: NIOAsyncChannel
+
+extension NIOPipeBootstrap {
+    /// Create the `PipeChannel` with the provided file descriptor which is used for both input & output.
+    ///
+    /// This method is useful for specialilsed use-cases where you want to use `NIOPipeBootstrap` for say a serial line.
+    ///
+    /// - Note: If this method returns normally, SwiftNIO will close `inputDescriptor` and `outputDescriptor`
+    ///         when the `Channel` becomes inactive. You _must not_ do any further operations `inputDescriptor` or
+    ///         `outputDescriptor`, including `close`.
+    ///         If this method throws, you still own the file descriptors and are responsible for
+    ///         closing them.
+    ///
+    /// - Parameters:
+    ///   - fileDescriptor: The _Unix file descriptor_ for the input & output.
+    ///   - backpressureStrategy: The back pressure strategy used by the channel.
+    ///   - isOutboundHalfClosureEnabled: Indicates if half closure is enabled on the channel. If half closure is enabled
+    ///   then finishing the `NIOAsyncChannelWriter` will lead to half closure.
+    ///   - inboundType: The channel's inbound type.
+    ///   - outboundType: The channel's outbound type.
+    /// - Returns: A `NIOAsyncChannel` for the bound connection.
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    @_spi(AsyncChannel)
+    public func withInputOutputDescriptor<Inbound: Sendable, Outbound: Sendable>(
```

Comment:
Hmm `with` suggests a block here which isn't the case (I see these are carried over from the existing bootstrap) -- I wonder if we should try to use a better name here?

Relatedly this takes ownership of the descriptor which is only clear from the docs. We should probably try to weave that into the name as well. I wonder if something like `takingOwnershipOfInputOutputDescriptor(...)` makes sense?

Reply:
cc @Lukasa @weissi WDYT?

Reply:
I like George's suggestion here.

Reply:
Do we want to deprecate and rename the old methods as well to align ELF and async APIs?

Reply:
Renamed all methods and deprecated the old ones